include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -152,6 +164,60 @@ inline AtomicType<std::complex<double> >::AtomicType()
 }
 
 
+// Calculate the padding required to align an element of a struct
+#define _STRUCT_PADDING(current_size, member_size) (((member_size) - (current_size)) % (member_size))
```

Comment:
I would prefix the `_STRUCT_PADDING` macro with `H5` to prevent conflicts and `#undef` it when not required anymore.